worker/worker.go

```diff
@@ -1390,9 +1386,18 @@ func (w *worker) Shutdown(ctx context.Context) error {
 	return nil
 }
 
-func (w *worker) scanHost(ctx context.Context, hostKey types.PublicKey, hostIP string) (rhpv2.HostSettings, rhpv3.HostPriceTable, time.Duration, error) {
+func (w *worker) scanHost(timeout time.Duration, hostKey types.PublicKey, hostIP string) (rhpv2.HostSettings, rhpv3.HostPriceTable, time.Duration, error) {
+	logger := w.logger.With("host", hostKey).With("hostIP", hostIP).With("timeout", timeout)
+
 	// prepare a helper for scanning
 	scan := func() (rhpv2.HostSettings, rhpv3.HostPriceTable, time.Duration, error) {
+		// apply timeout
+		ctx := w.shutdownCtx
```

Comment:
Instead of using the `shutdownCtx` as a parent, would it not be better to pass the request context as well as the timeout? If will eventually help towards a more graceful worker shutdown and I think we should respect a request context whenever we can, nothing that's more annoying than a request I can't cancel prematurely if I want to.